sdk/src/utils/merkle.rs

```diff
@@ -0,0 +1,118 @@
+// Copyright 2023 Adobe. All rights reserved.
+// This file is licensed to you under the Apache License,
+// Version 2.0 (http://www.apache.org/licenses/LICENSE-2.0)
+// or the MIT license (http://opensource.org/licenses/MIT),
+// at your option.
+
+// Unless required by applicable law or agreed to in writing,
+// this software is distributed on an "AS IS" BASIS, WITHOUT
+// WARRANTIES OR REPRESENTATIONS OF ANY KIND, either express or
+// implied. See the LICENSE-MIT and LICENSE-APACHE files for the
+// specific language governing permissions and limitations under
+// each license.
+
+use extfmt::Hexlify;
+
+use super::hash_utils::{concat_and_hash, hash_by_alg};
+use crate::{Error, Result};
+
+#[derive(Default, Clone, PartialEq, Debug)]
+pub struct MerkleNode(pub Vec<u8>);
+
+// Implements Merkle tree support corresponding to the C2PA spec variant.  The Merkle tree is not reduced and
+// all leaves live at the bottom most level.  If the last layer node is an odd index (lacking a matching pair),
+// its node value is propagated to parent layer, no cloning or hashing is expected.  Null tree entries do not contribute to the hashes.
+pub struct C2PAMerkleTree {
+    pub leaves: Vec<MerkleNode>,
+    pub layers: Vec<Vec<MerkleNode>>,
+}
+
+#[allow(dead_code)]
```

Comment:
Can we use a feature flag here?

Reply:
Why, we have to read these and will write them shortly.